src/DataReader/component_loader.rs

```diff
@@ -6,6 +6,27 @@ use crate::ModelObjects::queries::Query;
 use crate::ModelObjects::system_declarations::SystemDeclarations;
 use crate::System::input_enabler;
 use std::collections::HashMap;
+use std::sync::{RwLock, Arc};
+
+
+/// A struct used for caching the models sent with queries.
+#[derive(Debug, Default, Clone)]
+pub struct ModelCache {
+    Cache: Arc<RwLock<HashMap<String, Arc<ComponentContainer>>>>,
+}
+
+impl ModelCache {
+    pub fn get_model(&self, containerHash: &str) -> Arc<ComponentContainer> {
```

Comment:
This should return an optional.
If the components are not in the cache we want to insert them later

Reply:
done